lib/search/best_bets_checker.rb

```diff
@@ -9,16 +9,12 @@ def initialize(query, index = nil)
     end
 
     def best_bets
-      unless @fetched
-        fetch
-      end
+      @fetched || fetch
```

Comment:
It is a bit of a horrible pattern and I think you are right long term splitting the class would be a nicer solution, but was turning super hard to do small cleanup rather than a big refactor.

I have changed it so the `@fetched` check is inside the `fetch` method which I think is nicer